src/seedu/addressbook/AddressBook.java

```diff
@@ -544,6 +555,63 @@ private static String executeListAllPersonsInAddressBook() {
         showToUser(toBeDisplayed);
         return getMessageForPersonsDisplayedSummary(toBeDisplayed);
     }
+    
+    /**
+     * Displays all sorted persons in the address book to the user in asc/desc order.
+     *
+     * @return feedback display message for the operation result
+     */
+    private static String executeListAllPersonsInAddressBook(String commandArgs) {
+        if (!isSortPersonsArgValid(commandArgs)) {
+        	return getMessageForInvalidCommandInput(COMMAND_SORT_WORD, getUsageInfoForSortCommand());
+        }
+        final String extractedArg = extractCommandFromSortPersonsArg(commandArgs);
+        ArrayList<HashMap<PersonProperty, String>> personList = getAllPersonsInAddressBook();
+        sortAllPersonsInAddressBookAsc(personList, extractedArg);
+        showToUser(personList);
+        return getMessageForPersonsDisplayedSummary(personList);
+    }
+    
+    /**
+     * Extracts the command (asc or desc) from the raw sort args string
+     *
+     * @param rawArgs raw command args string for the sort person list command
+     * @return extracted arg
+     */
+    private static String extractCommandFromSortPersonsArg(String arg) {
+    	return arg.trim().toLowerCase();
+    }
+    
+    /**
+     * Checks validity of sort person list argument string's format.
+     *
+     * @param rawArgs raw command args string for the sort person list command
+     * @return whether the input args string is valid
+     */
+    private static boolean isSortPersonsArgValid(String rawArgs) {
+    	final String extractedArg = rawArgs.trim().toLowerCase();
+    	if (extractedArg.isEmpty()) {
+    		return false;
+    	}
+    	return true;
+    }
+    
+    /**
+     * Sorts all persons in the address book to the user.
+     *
+     * @return the sorted persons list in asc/desc order
+     */
+    private static ArrayList<HashMap<PersonProperty, String>> sortAllPersonsInAddressBookAsc(ArrayList<HashMap<PersonProperty, String>> unsortedPersonList,
+    		String arg) {
+    	if (arg.equals(COMMAND_SORT_PARAMETER_DESC)) {
+    		Collections.sort(unsortedPersonList, (HashMap<PersonProperty, String> person1, HashMap<PersonProperty, String> person2) 
+    				-> getNameFromPerson(person2).compareTo(getNameFromPerson(person1)));
```

Comment:
this way of splitting the line is not according to the coding standards we follow.
please take a look at the coding guidelines document and see how you can split a long line.